ldk-server/src/util/logger.rs

```diff
@@ -0,0 +1,123 @@
+use log::{Level, LevelFilter, Log, Metadata, Record};
+use std::fs::{self, File, OpenOptions};
+use std::io::{self, BufWriter, Write};
+use std::path::Path;
+use std::sync::Mutex;
+
+/// A logger implementation that writes logs to both stderr and a file.
+///
+/// The logger formats log messages with RFC3339 timestamps and writes them to:
+/// - stdout/stderr for console output
+/// - A file specified during initialization
+///
+/// All log messages follow the format:
+/// `[TIMESTAMP LEVEL TARGET] MESSAGE`
+///
+/// Example: `[2025-12-04T10:30:45Z INFO ldk_server] Starting up...`
+pub struct ServerLogger {
+	/// The minimum log level to display
+	level: LevelFilter,
+	/// The file to write logs to, buffered and protected by a mutex for thread-safe access
+	file: Mutex<BufWriter<File>>,
+}
+
+impl ServerLogger {
+	/// Initializes the global logger with the specified level and file path.
+	///
+	/// Opens or creates the log file at the given path. If the file exists, logs are appended.
+	/// If the file doesn't exist, it will be created along with any necessary parent directories.
+	///
+	/// This should be called once at application startup. Subsequent calls will fail.
+	pub fn init(level: LevelFilter, log_file_path: &Path) -> Result<(), io::Error> {
+		// Open or create the log file
+		let file = if log_file_path.exists() {
+			OpenOptions::new().append(true).open(log_file_path)?
```

Comment:
Why not just always call `create_dir_all` (which is idempotent) and then use `fs::OpenOptions::new().create(true).append(true)`?

Reply:
TIL